yt_dlp/extractor/soundcloud.py

```diff
@@ -112,21 +111,29 @@ def _download_json(self, *args, **kwargs):
     def _initialize_pre_login(self):
         self._CLIENT_ID = self.cache.load('soundcloud', 'client_id') or 'a3e059563d7fd3372b49b37f00a00bcf'
 
+    def _verify_oauth_token(self, token):
+        if self._request_webpage(
+                self._API_VERIFY_AUTH_TOKEN % (self._API_AUTH_QUERY_TEMPLATE % self._CLIENT_ID),
+                None, note='Verifying login token...', fatal=False,
+                data=json.dumps({'session': {'access_token': token}}).encode()):
+            self._HEADERS = {'Authorization': f'OAuth {token}'}
+            self.report_login()
+        else:
+            self.report_warning('Provided authorization token seems to be invalid. Continuing as guest')
+
+    def _real_initialize(self):
+        if self._HEADERS:
+            return
```

Comment:
In current impl, this is unreachable, no?

Reply:
`_perform_login` is called before `_real_initialize`, right? this prevents a double verification after login with `-u oauth -p token`